### DISCUSSION TITLE
custom files

### DISCUSSION THREAD

**Discussion by ntrits** [make-all/tuya-local#779] [Q&A]
Can I add optional files to the devices folder and not lose them when upgrading?

**Comment by make-all** [comment]
If you update via HACS, I think it removes the entire directory when upgrading.
If you manually install you can manage the files how you wish.

**Comment by ntrits** [comment]
Thanks for reply.
Upgrade via HACS.
Could there be some way to not delete all folders or to include some custom folder?
Now every time there is an upgrade I have to copy the extra files I created to the devices folder.

Example: https://github.com/ntrits/tuya-local-devices

**Reply by make-all** [reply]
No, it is not my goal to help people keep their own private device configs for themselves.

**Comment by ntrits** [comment]
OK, that's a very good point.
Can these be integrated as well?
https://github.com/ntrits/tuya-local-devices

And what should I do so they can be integrated with any others I create in the future?

**Comment by make-all** [comment]
smartmcb_smt006_energymeter.yaml already exists, so I am looking though your config to see what the differences are.  For the other one, you should submit a pull request or new device request with all the details.

**Comment by ntrits** [comment]
Thank you!

**Comment by rac146** [comment]
I'm curious why we can't have support for a separate folder to keep custom files outside of the custom_components directory? I have a few I use for testing and lost them after an upgrade.
